raiden/network/rpc/client.py

```diff
@@ -850,7 +937,10 @@ def send_transaction(self) -> TransactionHash:
 
                     try:
                         tx_hash = client.web3.eth.sendRawTransaction(signed_txn.rawTransaction)
+                        self._sent = TransactionSlotState.sent
                     except ValueError as e:
+                        self._sent = TransactionSlotState.rejected
```

Comment:
This may not be 100% correct, since it assumes that `web3` will always raise an exception if the transaction is not added to the transaction pool ... however that could happen asynchronously, or the transaction could be removed from the pool because of a bug (happened in the past). But this should be good enough for now.